src/v/datalake/schema_registry.cc

```diff
@@ -24,4 +25,36 @@ get_schema_id_result get_value_schema_id(const iobuf& buf) {
     return pandaproxy::schema_registry::schema_id{id};
 }
 
+// TODO: this is mostly a copy-and-paste of get_proto_offsets from
+// pandaproxy::schema_registry with a slightly different interface. Unify these.
+get_proto_offsets_result get_proto_offsets(const iobuf& buf) {
+    // The encoding is a length, followed by indexes into the file or message.
+    // Each number is a zigzag encoded integer.
+    iobuf_const_parser parser(buf);
+    // skip schema id prefix
+    parser.skip(sizeof(uint8_t) + sizeof(int32_t));
+    std::vector<int32_t> offsets;
+    auto [offset_count, bytes_read] = parser.read_varlong();
+    if (!bytes_read) {
+        return get_schema_error::bad_varint;
+    }
+    // Reject more offsets than bytes remaining; it's not possible
+    if (static_cast<size_t>(offset_count) > parser.bytes_left()) {
+        return get_schema_error::not_enough_bytes;
+    }
+    offsets.resize(offset_count);
+    for (auto& o : offsets) {
+        std::tie(o, bytes_read) = parser.read_varlong();
+        if (!bytes_read) {
+            return get_schema_error::bad_varint;
+        }
+    }
+    if (offsets.empty()) {
```

Comment:
do you also need to validate that the number of offsets read is the same as offset_count?

Reply:
I'm not sure how they wouldn't be. The loop that reads the offsets will execute that many times, and if it fails to read an offset it returns an error.

Reply:
Invalid offsets aside, if `offset_count == 0` then we should return `{0}`. Here's the same method in franz-go:

https://github.com/twmb/franz-go/blob/b77dd13e2bfaee7f5181df27b40ee4a4f6a73b09/pkg/sr/serde.go#L482-L507

We do that here, but in a very roundabout way by resizing then checking the result of that method.

Reply:
>I'm not sure how they wouldn't be

i'm asking if it is safe to trust the size value you are decoding from the wire format as the prefix on the data.

Reply:
?

Reply:
There is a test for this case, and generally our iobuf parser throws on short input